linux_new/working_v3.groovy

```diff
@@ -0,0 +1,795 @@
+                    //                   (arch == 'arm') ? 'docker&&linux&&aarch64' : 
+                    //                   (arch == 'aarch64') ? 'docker&&linux&&aarch64' : 
+                    //                   (arch == 'ppc64le') ? 'build&&docker&&ppc64le' :
+                    //                   (arch == 's390x') ? 'docker&&s390x&&dockerBuild' :
+                    //                   (arch == 'riscv64') ? 'dockerbuild&&linux&&riscv64' :
+                    //                   'build&&linux&&x64&&dockerBuild&&dynamicAzure'
+                    
+                    // PKGBUILDLABELRHEL = 'build&&linux&&x64&&dockerBuild&&dynamicAzure'
+
+                    packagearchDeb = (arch == 'x64') ? 'amd64' :
+                                     (arch == 'arm') ? 'armhf' :
+                                     (arch == 'aarch64') ? 'arm64' :
+                                     (arch == 'ppc64le') ? 'ppc64el' :
+                                     (arch == 's390x') ? 's390x' :
+                                     (arch == 'riscv64') ? 'riscv64' :
+                                     'unknown'
+
+                    // Fix Architectures To Be RHEL/Suse Control File Compatible
+                    // s390x and riscv64 are OK
+                    // Reqd: x64 armv7hl aarch64 ppc64le
+
+                    packagearchRhel = (arch == 'x64') ? 'x86_64' :
+                                      (arch == 'arm') ? 'armv7hl' :
+                                      (arch == 'aarch64') ? 'aarch64' :
+                                      (arch == 'ppc64le') ? 'ppc64le' :
+                                      (arch == 's390x') ? 's390x' :
+                                      (arch == 'riscv64') ? 'riscv64' :
+                                      'unknown'
+
+
+                    // Version Number Parsing
+
+                    def vername = ''
+                    def verversion = ''
+                    def verbuild = ''
+
+                    // Add Special Handling For JDK8 Version Number
+                    // jdk8u432-b06
+
+                    if (params.Tag?.startsWith("jdk8")) {
+                        echo "JDK 8"
+                        def split = (params.Tag =~ /(jdk)(\d+u\d+)-(b\d+)/)
+                        if (split.find()) {
+                            // println("Full Match: ${split[0][0]}")
+                            vername = split[0][1]          // "jdk"
+                            verversion = split[0][2]       // "8u432"
+                            verbuild = split[0][3]         // "06"
+                            } else {
+                                 error("The version string format does not match the expected pattern.")
+                        }
+                    } else {
+                        echo "Not JDK8"
+                    // Parse The Version Tag, Into Usable Components
+                    
+                    def split = (params.Tag =~ /(jdk)-(\d[\d.]*)([+_]\d+)?/)
```

Comment:
Does this need to be an escaped dot? Assuming yes, because otherwise the square brackets seem superfluous.

```suggestion
                    def split = (params.Tag =~ /(jdk)-(\d[\d\.]*)([+_]\d+)?/)
```

Reply:
File removed, but yes,t the regex is an odd one :)